### PR TITLE
support both serialized protos and YAML

### PR DESCRIPTION
**Pull request by theganyo**
Task 2 in #946: Update registry subcommands that read artifacts to support both serialized protos and YAML.

### CODE REVIEW

**Comment by theganyo** [comment]
A note on my testing this: One interesting test I did was to run the full battery of tests after forcing the default to yaml storage. I say interesting because everything works as expected... except the conformance tests:

```
--- FAIL: TestConformance (0.19s)
    --- FAIL: TestConformance/normal_case (0.05s)
        conformance_test.go:465: Failed getting artifact contents projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest: rpc error: code = NotFound desc = "projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest" not found in database
    --- FAIL: TestConformance/default_case (0.04s)
        conformance_test.go:465: Failed getting artifact contents projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-default: rpc error: code = NotFound desc = "projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-default" not found in database
    --- FAIL: TestConformance/multiple_severity (0.03s)
        conformance_test.go:465: Failed getting artifact contents projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-multiple-severity: rpc error: code = NotFound desc = "projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-multiple-severity" not found in database
    --- FAIL: TestConformance/multiple_state (0.04s)
        conformance_test.go:465: Failed getting artifact contents projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-multiple-state: rpc error: code = NotFound desc = "projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-multiple-state" not found in database
    --- FAIL: TestConformance/multiple_linter (0.04s)
        conformance_test.go:465: Failed getting artifact contents projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-multiple-linter: rpc error: code = NotFound desc = "projects/conformance-test/locations/global/apis/petstore/versions/1.0.0/specs/openapi/artifacts/conformance-openapitest-multiple-linter" not found in database
```

I couldn't identify why these were failing as the conformance tests aren't really unit tests - they start processes and move data around in ways that are hidden from the test suite. In addition, nothing was logged that indicated what the root cause might be.

In any case, this maybe isn't something that necessarily needs to be addressed within this PR, but it is something that will need to be addressed before we can finish #946.

**Comment by timburks** [comment]
> after forcing the archive storage to default to yaml

Just wondering... how did you do that? By temporarily hard-coding 
[this](https://github.com/apigee/registry/blob/8dab302177542a97f479154ae1079c8385cb8b95/cmd/registry/cmd/apply/apply.go#L32) to true?

File: cmd/registry/patch/patch_test.go [hunk 1]
```diff
@@ -1009,64 +1011,82 @@ func TestMessageArtifactPatches(t *testing.T) {
...
-			}
-			err = visitor.GetArtifact(ctx, registryClient, artifactName, true,
-				func(ctx context.Context, artifact *rpc.Artifact) error {
-					contents, err := getArtifactMessageContents(artifact)
+	storageTypes := []struct {
+		mimeBase string
+		ctx      context.Context
+	}{
+		{"application/octet-stream", context.Background()},
+		{"application/yaml", SetStoreArchivesAsYaml(context.Background())},
```
Comment (timburks) [inline cmd/registry/patch/patch_test.go:1020]: This is our main test that this works, right? Verifying that all artifact types can be stored and read in both formats? (just thinking out loud... LGTM)

**Review by timburks — APPROVED** [review]
I'm curious about the conformance test problem, but that's not breaking anything since we're still using proto storage and testing everything with that. LGTM!

File: cmd/registry/patch/patch_test.go (hunk 1 above)
Comment (theganyo) [reply cmd/registry/patch/patch_test.go:1020]: > > after forcing the archive storage to default to yaml
> 
> Just wondering... how did you do that? By temporarily hard-coding [this](https://github.com/apigee/registry/blob/8dab302177542a97f479154ae1079c8385cb8b95/cmd/registry/cmd/apply/apply.go#L32) to true?

Close, but [here](https://github.com/apigee/registry/blob/8dab302177542a97f479154ae1079c8385cb8b95/cmd/registry/patch/artifact.go#L113)

**Comment by theganyo** [comment]
> > after forcing the archive storage to default to yaml
> 
> Just wondering... how did you do that? By temporarily hard-coding [this](https://github.com/apigee/registry/blob/8dab302177542a97f479154ae1079c8385cb8b95/cmd/registry/cmd/apply/apply.go#L32) to true?

Close, but [here](https://github.com/apigee/registry/blob/8dab302177542a97f479154ae1079c8385cb8b95/cmd/registry/patch/artifact.go#L113)

File: cmd/registry/patch/patch_test.go (hunk 1 above)
Comment (theganyo) [reply cmd/registry/patch/patch_test.go:1020]: Yes

**Pull request merged**